### PR TITLE
Remove BC Layer for Symfony Dispatcher versions 4.3 and lower

### PR DESCRIPTION
**Pull request by rpkamp**

### CODE REVIEW

**Comment by rpkamp** [comment]
This PR partially resolves https://github.com/Behat/Behat/issues/1305

**Comment by rpkamp** [comment]
Scrutenizer errors not introduced by this PR, they were already there.

File: src/Behat/Testwork/EventDispatcher/TestworkEventDispatcherSymfonyLegacy.php [hunk 1]
```diff
@@ -13,6 +13,9 @@
  * if the old symfony interface is detected.
  *
  * @deprecated Do not reference this class directly, use TestworkEventDispatcher
+ *
+ * @todo Remove this class in the next major version. It's not used anymore, but just left here
+ *       for BC purposes.
```
Comment (Taluu) [inline src/Behat/Testwork/EventDispatcher/TestworkEventDispatcherSymfonyLegacy.php:18]: add some `trigger_error` instead of such comments

Comment (rpkamp) [reply src/Behat/Testwork/EventDispatcher/TestworkEventDispatcherSymfonyLegacy.php]: Good call, I've removed the comment and added some `trigger_error` with `E_USER_DEPRECATED` 👍

**Comment by ciaranmcnulty** [comment]
I think longer term we should be looking at decoupling from the symfony interfaces here, but this looks fine

**Comment by rpkamp** [comment]
Anything needed from me here to push this forward?

**Comment by ciaranmcnulty** [comment]
Thanks!

**Pull request merged**